modules/apps/object/object-service/src/main/java/com/liferay/object/internal/model/listener/UserModelListener.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.object.internal.model.listener;
+
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.system.SystemObjectDefinitionManager;
+import com.liferay.object.system.SystemObjectDefinitionManagerRegistry;
+import com.liferay.portal.kernel.exception.ModelListenerException;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.model.BaseModelListener;
+import com.liferay.portal.kernel.model.ModelListener;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Carolina Barbosa
+ */
+@Component(service = ModelListener.class)
+public class UserModelListener extends BaseModelListener<User> {
+
+	@Override
+	public void onAfterUpdate(User originalUser, User user)
+		throws ModelListenerException {
+
+		if (!user.isGuestUser()) {
+			return;
+		}
```

Comment:
Question: Do we need to update only when the user is a guest?

Reply:
Yest, because the guest user it's the company user. If I change the company default locale, only the guest user will be updated.